app/src/main/java/ch/epfl/sdp/drone3d/ui/MainActivity.kt

```diff
@@ -114,4 +119,17 @@ class MainActivity : AppCompatActivity() {
         val intent = Intent(this, activity.java)
         startActivity(intent)
     }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        permissionService.onRequestPermissionsResult(
+            requestCode,
+            permissions as Array<String>,
```

Comment:
With the other changes, you can remove the 'as Array<String>' as no cast is needed